bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,797 @@
+            logger.warn("air-Q - airqHandler - encrypt(): Error while encrypting: {}", e.toString());
+        }
+        return content;
+    }
+
+    // gets the data after online/offline management and does the JSON work, or at least the first step.
+    protected @Nullable String getDecryptedContentString(String url, String requestMethod, @Nullable String body) {
+        Result res = null;
+        String jsonAnswer = null;
+        res = getData(url, "GET", null);
+        if (res != null) {
+            String jsontext = res.getBody();
+            logger.trace("air-Q - airqHandler - getDecryptedContentString(): Result from getData() is {} with body={}",
+                    res, res.getBody());
+            // Gson code based on https://riptutorial.com/de/gson
+            Gson gson = new Gson();
+            JsonElement ans = gson.fromJson(jsontext, JsonElement.class);
+            if (ans != null) {
+                JsonObject jsonObj = ans.getAsJsonObject();
+                jsonAnswer = decrypt(jsonObj.get("content").getAsString().getBytes(), config.password);
+                if (jsonAnswer == null) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                            "Decryption not possible, probably wrong password");
+                }
+            } else {
+                logger.warn(
+                        "air-Q - airqHandler - getDecryptedContentString(): The air-Q data could not be extracted from this string: {}",
+                        ans);
+            }
+        }
+        return jsonAnswer;
+    }
+
+    // calls the networking job and in addition does additional tests for online/offline management
+    protected @Nullable Result getData(String address, String requestMethod, @Nullable String body) {
+        Result res = null;
+        res = doNetwork(address, "GET", body);
+        if (res == null) {
+            if (thStatus != ThingStatus.OFFLINE) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "air-Q device not reachable");
+                thStatus = ThingStatus.OFFLINE;
+            } else {
+                logger.warn("air-Q - airqHandler - getData(): retried but still cannot reach the air-Q device.");
+            }
+        } else {
+            if (thStatus == ThingStatus.OFFLINE) {
+                updateStatus(ThingStatus.ONLINE);
+                thStatus = ThingStatus.ONLINE;
+            }
+        }
+        return res;
+    }
+
+    // does the networking job (and only that)
+    protected @Nullable Result doNetwork(String address, String requestMethod, @Nullable String body) {
+        int timeout = 10000;
+        HttpURLConnection conn = null;
+        logger.debug("air-Q - airqHandler - doNetwork(): connecting to {} with method {} and body {}", address,
+                requestMethod, body);
+        try {
+            conn = (HttpURLConnection) new URL(address).openConnection();
```

Comment:
You can save resources by using the framework's `HttpClient`. See https://github.com/openhab/openhab-addons/search?q=httpclientfactory for example.